flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/utils/SourceCodeGenerator.java

```diff
@@ -43,6 +44,87 @@ public String generateCodeSnippetForType(TypeData typeData) {
         };
     }
 
+    public String generateGraphqlClassType(TypeData typeData) {
+        NodeKind nodeKind = typeData.codedata().node();
```

Comment:
Instead of using completely depending on StringBuilder with a lot of manual `.append()` calls, we can alternatively use java String formatting or Text Blocks (along with StringBuilder) for better readability and less complexity.